### DISCUSSION TITLE
openapi vs swagger

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#230] [Spring 뒷고기]
.

**Comment by gyungchan-jo** [comment]
### OpenAPI

OpenAPI는 OpenAPI Specification(OAS)라고 부르는데, 이는 API의 엔드포인트, 메서드, 요청/응답 형식, 파라미터 등을 정의합니다. API spec을 YAML 또는 JSON 형식으로 작성되며, API의 구조와 기능을 명확하게 서술할 수 있습니다.

### Swagger
Swagger는 OpenAPI를 기반으로 한 도구 모음입니다. Swagger는 API 문서화를 쉽게 하고, API의 기능을 테스트할 수 있는 다양한 툴을 제공합니다. Swagger 프로젝트는 OpenAPI 사양을 작성, 시각화, 테스트, 관리하는 데 필요한 다양한 도구를 포함합니다.

주요 도구
- Swagger Editor: 브라우저 기반의 편집기, OpenAPI Spec을 쉽게 작성할 수 있게 도와줍니다.
- Swagger UI: OpenAPI spec문서를 브라우저에서 확인할 수 있게 해줍니다. API Test도 가능합니다.
- Swagger Codegen: OpenAPI spec에 맞게 Server나 Client의 stub code 생성합니다.
